pulp_smash/resources/platform/api_v2/api.py

```diff
@@ -0,0 +1,183 @@
+        self.cfg = get_config()
+
+    def create_repo(self, **kwargs):
+        """Create repository on pulp server.
+        After calling this method, <repo>.last_response.raise_for_status()
+        should be called in order to make sure that repo was correctly created.
+        :param kwargs: Additional arguments which will be passed to request,
+        same as in :class:`Repository` constructor.
+        """
+        self.data_keys.update(kwargs)
+        self.last_response = requests.post(
+            self.cfg.base_url + CREATE_REPOSITORY_PATH,
+            json=self.data_keys,
+            **self.cfg.get_requests_kwargs()
+        )
+
+    def delete_repo(self):
+        """Delete repository from pulp server.
+        After calling this method, <repo>.last_response.raise_for_status()
+        Taks.wait_for_tasks(<repo>.last_response) should be called in order
+        to make sure repo was correctly deleted.
+        """
+        self.last_response = requests.delete(
+            self.cfg.base_url +
+            REPOSITORY_PATH.format(self.data_keys['id']),
+            **self.cfg.get_requests_kwargs()
+        )
+
+    def get_repo(self):
+        """Get information about repository on server.
+        After calling this method, <repo>.last_response.raise_for_status()
+        should be called in order to make sure that call was succesfull.
+        """
+        self.last_response = requests.get(
+            self.cfg.base_url + REPOSITORY_PATH.format(self.data_keys['id']),
+            **self.cfg.get_requests_kwargs()
+        )
+
+    def update_repo(self,
+                    delta,
+                    importer_config=None,
+                    distributor_configs=None):
+        """Update repository with keys from kwargs.
+        After calling this method, <repo>.last_response.raise_for_status()
+        and Task.wait_for_tasks(<repo>.last_response)
+        should be called in order to make sure repo was correctly updated.
+        :param delta: Object containing keys with values that should
+            be updated on the repository.
+        :param importer_config: Object containing keys with values that should
+            be updated on the repository’s importer config.
+        :param distributor_configs: object containing keys that
+            are distributor ids
+        """
+        my_delta = {'delta': delta}
+        if importer_config is not None:
+            my_delta.update({'importer_config': importer_config})
+        if distributor_configs is not None:
+            my_delta.update({'distributor_configs': distributor_configs})
+        self.last_response = requests.put(
+            self.cfg.base_url + REPOSITORY_PATH.format(self.data_keys['id']),
+            json=my_delta,
+            **self.cfg.get_requests_kwargs()
+        )
+
+
+class Task(object):
+    """Handles tasks related operations. So far only waiting for given tasks
+    to immediate finish is implemented.
+    """
+    cfg = get_config()
```

Comment:
In my experience, class attributes are deceptive. They seem simple when you start using them, but become a real pain down the road when more use cases for your class arise. Here's two specific issues:

* Let's say a user installs Pulp Smash for the very first time and executes `make docs-html`. As part of the documentation generation process, this module is imported and `get_config` is called. Since the user has just installed Pulp Smash, a config will not be found, and a `ConfigFileNotFoundError` is raised. :cry:
* Let's say we have two Pulp servers, and we want to monitor a task on each of them. It would be nice if we could instantiate two `Task` objects, give each a unique `cfg` object and call `wait_for_task()` on each. Using a class attribute makes this impossible.